jdaviz/configs/imviz/plugins/orientation/orientation.py

```diff
@@ -351,36 +354,54 @@ def add_orientation(self, rotation_angle=None, east_left=None, label=None,
         )
 
         # add orientation layer to all viewers:
-        self._add_data_to_all_viewers(label)
+        for viewer_ref in self.app._viewer_store:
+            self._add_data_to_viewer(label, viewer_ref)
 
         if set_on_create:
-            self.orientation._update_layer_items()
             self.orientation.selected = label
 
-    def _add_data_to_all_viewers(self, data_label):
-        for viewer_ref in self.app.get_viewer_reference_names():
-            layers = [
-                layer.label for layer in
-                self.app.get_viewer(viewer_ref).layers
-            ]
-            if data_label not in layers:
-                self.app.add_data_to_viewer(viewer_ref, data_label)
+    def _add_data_to_viewer(self, data_label, viewer_id):
+        viewer = self.app.get_viewer_by_id(viewer_id)
+
+        wcs_only_layers = viewer.state.wcs_only_layers
+        if data_label not in wcs_only_layers:
+            self.app.add_data_to_viewer(viewer_id, data_label)
 
     def _on_viewer_added(self, msg):
-        for data_label in self.orientation.choices:
-            self._add_data_to_all_viewers(data_label)
+        self._send_wcs_layers_to_all_viewers(viewers_to_update=[msg._viewer_id])
+
+    @observe('viewer_items')
+    def _send_wcs_layers_to_all_viewers(self, *args, **kwargs):
+        if not hasattr(self, 'viewer'):
+            return
+
+        wcs_only_layers = self.app.get_viewer_by_id('imviz-0').state.wcs_only_layers

```

Comment:
should this use the `default_image_viewer` thing instead of hardcoding `imviz-0`?

Reply:
Yeah, something like `self.app._jdaviz_helper.default_viewer.state.wcs_only_layers` would work, I think.

Reply:
Works for me.
```suggestion
        wcs_only_layers = self.app._jdaviz_helper.default_viewer.state.wcs_only_layers
```

Reply:
Does this work still?  That might expose the public API, so you might need `._obj` in there now (if you have https://github.com/spacetelescope/jdaviz/pull/2563, and otherwise will need it on rebase).

Reply:
that's right – I pushed a correction.